lib/ClangImporter/ClangImporter.cpp

```diff
@@ -4005,6 +4004,43 @@ void ClangImporter::Implementation::lookupAllObjCMembers(
   }
 }
 
+SmallVector<SwiftLookupTable::SingleEntry, 4>
+ClangDirectLookupRequest::evaluate(Evaluator &evaluator,
+                                   ClangDirectLookupDescriptor desc) const {
+  auto &ctx = desc.decl->getASTContext();
+
+  auto *clangDecl = desc.decl->getClangDecl();
+  auto clangModule = getClangSubmoduleForDecl(clangDecl);
+  auto *lookupTable = ctx.getClangModuleLoader()->findLookupTable(*clangModule);
+
+  auto *swiftDeclContext = desc.decl->getInnermostDeclContext();
+  auto *declContextTypeDecl = swiftDeclContext->getSelfNominalTypeDecl();
+  auto effectiveClangContext =
+      ctx.getClangModuleLoader()->getEffectiveClangContext(declContextTypeDecl);
+
+  return lookupTable->lookup(SerializedSwiftName(desc.name.getBaseName()),
+                             effectiveClangContext);
+}
+
+// REVIEW-QUESTION: does it make sense for this to be its own request? It's
+// modifying global state via the importDeclDirectly call, so that's why I made
+// it a request. If this is the correct approach, I'll pull more of the
+// ImportDecl visitors into their own requests, like I do here for namespaces.
```

Comment:
This makes perfect sense to me, although I'm not super familiar with the Request API design. If I'm understanding it correctly, the ideal state would be to eventually migrate all of the ClangImporter to Requests, right? I that case, I think it makes sense to have these visitor invocations as separate requests.